src/aws-cpp-sdk-core/source/client/AWSClient.cpp

```diff
@@ -774,61 +774,101 @@ void AWSClient::AppendHeaderValueToRequest(const std::shared_ptr<Aws::Http::Http
 void AWSClient::AddChecksumToRequest(const std::shared_ptr<Aws::Http::HttpRequest>& httpRequest,
     const Aws::AmazonWebServiceRequest& request) const
 {
+    //check if user has provided the checksum algorithm
     Aws::String checksumAlgorithmName = Aws::Utils::StringUtils::ToLower(request.GetChecksumAlgorithmName().c_str());
+    bool checksumAlgorithmProvided = !checksumAlgorithmName.empty();
 
     // Request checksums
-    if (!checksumAlgorithmName.empty())
+    if (checksumAlgorithmProvided)
     {
+        // Check if user has provided a checksum value for the specified algorithm
+        Aws::String checksum_type = "x-amz-checksum-" + checksumAlgorithmName;
```

Comment:
this file (and SDK) uses camelCase.
also consider adding const if this variable is not updated later